src/WatchKit/iOS/WKUserNotificationInterfaceController.cs

```diff
@@ -45,7 +47,7 @@ public virtual void DidReceiveRemoteNotification (NSDictionary remoteNotificatio
 			throw new PlatformNotSupportedException (Constants.WatchKitRemoved);
 		}
 
-		public virtual void DismissController ()
+		public new virtual void DismissController ()
```

Comment:
This is surprising. What's the story here?

Reply:
@chamons Good question, the compiler told me that I should and I kind of understood so I did, but because you asked, I read more on it and understand it a little better now!
The `WKUserNotificationInterfaceController` class inherits from `WKInterfaceController` and both classes contain a method definition for `DismissController`.  The `WKUserNotificationInterfaceController` version above does not contain an 'override' keyword so implicitly, the compiler is hiding  the base class version of `DismissController` inside `WKInterfaceController`. The compiler gives us a warning saying that it is already doing this, but we should mark it with the `new` keyword to be explicit. In this case however, both classes have the same definition for `DismissController` so I think adding a `new` or an `override` would give us the same results and perhaps we could even remove this method altogether? 

That last part I am not 100% sure though and would like to know if others know if there would be any effects if we remove this method. If we have `BaseClass`, `Derived1Class : BaseClass`, and `Derived2Class : Derived1Class` and BaseClass and Derived1Class both contain different method implementations for Method1 (), if we call base.Method1 () from Derived2Class, will it call BaseClass.Method1 () or Derived1Class.Method1 ()?

Reply:
Perhaps this would be good for me to bring to office hours ^

Reply:
You can't remove the method, because that's a breaking change.

In this case adding the `new` keyword is the correct thing to do, because it preserves the old behavior (the only change with the `new` keyword is that it silences the warning).

Changing to `override` would change the method to override the base method, which while it would probably be the correct thing to do from a design stand point, the fact is that this is deprecated API that doesn't even exist in the OS anymore, so our only concern is to not break backwards compatibility.

Reply:
Long thread.. so I tested it out and found out that in my new scenario, the Derived2Class's `base` will refer to Derived1Class!